numba/tests/test_parfors.py

```diff
@@ -2130,6 +2130,16 @@ def test_impl(a, b, size):
         self.assertEqual(countParfors(test_impl, cptypes), 2)
         self.check(test_impl, a, b, size)
 
+    def test_prange_optional(self):
+        @njit("f8[:](optional(u1))", parallel=True)
+        def foo(pred=None):
+            arr = np.ones(10)
+            for i in prange(1):
+                if pred is not None:
+                    arr[i] = 0.0
+            return arr
+        self.assertEqual(foo().sum(), 10.0)
```

Comment:
@apmasell Perhaps take a look at the first code sample in: https://github.com/numba/numba/issues/8453#issuecomment-1253372598, this function needs to be called with an `types.Optional`, a default kwarg goes through as `types.Omitted`.

Also, this should use the `self.check` functionality to assert the validity of the result.